### ISSUE TITLE
ArgumentException thrown when using the windows emoji picker

### ISSUE THREAD

**Issue by Orongto** [opentk/opentk#1363]
### Description

Using the windows emoji picker in an OpenTK window the application throws an ArgumentOutOfRangeException when processing window events.

### Repro steps

1. Open any OpenTK application using NuGet version 4.6.7. (unsure about other versions)
2. Open the windows emoji picker by holding windows key + full stop key. 
3. Choose any emoji. 
4. Close the emoji picker.

### Expected behavior

OpenTK does not crash.

### Actual behavior
Exception gets thrown when exiting emoji window:
```
System.ArgumentOutOfRangeException: A valid UTF32 value is between 0x000000 and 0x10ffff, inclusive, and should not include surrogate codepoint values (0x00d800 ~ 0x00dfff). (Parameter 'utf32')
   at System.Char.ConvertFromUtf32(Int32 utf32)
   at OpenTK.Windowing.Common.TextInputEventArgs.get_AsString()
   at Nurose.OpenTK.OpenTKInput.WindowBase_TextInput(TextInputEventArgs obj)
   at OpenTK.Windowing.Desktop.NativeWindow.OnTextInput(TextInputEventArgs e)
   at OpenTK.Windowing.Desktop.NativeWindow.<.ctor>b__125_5(Window* w, UInt32 codepoint)
   at OpenTK.Windowing.GraphicsLibraryFramework.GLFWNative.glfwPollEvents()
   at OpenTK.Windowing.Desktop.NativeWindow.ProcessEvents()
 ```

**Comment by g7ChoGXh** [comment]
It seem like the `uint codepoint` in `NativeWindow.CharCallback` will receive a sequence of UTF-16 code units instead of actual UTF-32 code points. For example, entering the woman facepalming emoji (🤦‍♀️) through the described procedure will produce calls to `CharCallback` with the following sequence of arguments for `codepoint` (base 10, tested with OpenTK 4.7.1 on Windows 10):
```
55358 (High Surrogate)
56614 (Low Surrogate, together Face Palm)
8205 (Zero Width Joiner)
9792 (Female Sign)
65039 (Variation Selector-16)
```
The surrogate code unit will then be passed to `TextInputEventArgs` which assumes it to be a non-surrogate UTF-32 code point by passing it to `char.ConvertFromUtf32`.

This behaviour seems to contradict the (somewhat unclear?) GLFW documentation at https://www.glfw.org/docs/latest/group__input.html#ga1ab90a55cf3f58639b893c0f4118cb6e.

I suggest that `NativeWindow` checks whether a high surrogate was passed to `CharCallback` and if so, stores it until the next low surrogate arrives. Then, the UTF-32 code point can be reconstructed and passed to `TextInputEventArgs`. If `CharCallback` would be called as documented, we should never see a surrogate character anyways, so the callback would not behave differently.

**Comment by utkumaden** [comment]
Nice catch. Normally the .NET library has a specific exception related to surrogate pairs, but since we are getting this from C, nobody must have run into the issue until an emoji was sent haha.

I think your solution will work.

**Comment by g7ChoGXh** [comment]
I would implement the fix myself, but I have no idea how to build and test OpenTK. Can anyone help me with that?

**Comment by utkumaden** [comment]
If you wish to implement it, join our discord where we can help you in #development. Building OpenTK is a error prone so you may need help. It boils down to running `build.cmd`. (when it works properly)

**Comment by NogginBops** [comment]
This is fixed in https://github.com/glfw/glfw/issues/1635, which should be solved with #1426.
